tasks/pkg-debian.yml

```diff
@@ -31,21 +36,28 @@
   with_items:
     - "deb http://apt.datadoghq.com/ stable main"
     - "deb http://apt.datadoghq.com/ stable 6"
+    - "deb http://apt.datadoghq.com/ stable 7"
   when: not ansible_check_mode
 
 - name: Ensure Datadog repository is up-to-date
   apt_repository:
-    repo: "{{ datadog_apt_repo }}"
-    state: "{% if datadog_agent5 %}absent{% else %}present{% endif %}"
+    filename: ansible_datadog_agent
+    repo: "{{ item.value }}"
+    state: "{% if item.key == datadog_agent_major_version|int and datadog_apt_repo | length == 0 %}present{% else %}absent{% endif %}"
```

Comment:
does this actually work if Agent v5 or v6 is selected? If I understand the logic correctly, using `with_dict` will create 3 tasks (one for each major version), which will in turn either delete or create the `/etc/apt/sources.list.d/ansible_datadog_agent.list` file. So I'd expect the last defined task to always "win", i.e. always the v7 task since it's the last item of `with_dict`.

Instead of having this logic in `state`, shouldn't the logic be in the `when` condition, and `state` always be set to `present`? I believe `when` is evaluated for each item of `with_dict`, but I may be wrong or missing something.

Reply:
As mentioned in the docs: https://docs.ansible.com/ansible/latest/modules/apt_repository_module.html, the state only applies to the source string, not the file: if `absent`, ansible will make sure that the `repo` string is not present in any source file: the file itself won't be erased.

Reply:
ok, thanks for the explanation 👍